cidc_api/models/models.py

```diff
@@ -606,6 +646,76 @@ def generate_sample_csv(cls, trial_id: str, session: Session) -> str:
             raise NoResultFound(f"No trial found with id {trial_id}")
         return unprism.unprism_samples(trial.metadata_json)
 
+    file_bundle: Optional[FileBundle]
+
+    @classmethod
+    @with_default_session
+    def list_with_file_bundles(cls, session: Session, **pagination_args):
+        """TODO: docstring"""
+        # Build a query that collects trials along with all files for that trial
+        query = (
+            session.query(
+                TrialMetadata,
+                # Aggregate files for this trial record as a JSON object with shape:
+                # [
+                #   {'type': <file1 type>, 'purpose': <file1 purpose>, 'id': <file1 id>},
+                #   {'type': <file2 type>, 'purpose': <file2 purpose>, 'id': <file2 id>},
+                #   ...
+                # ]
+                func.json_agg(
+                    func.json_build_object(
+                        "type",
+                        DownloadableFiles.consolidated_upload_type,
+                        "purpose",
+                        DownloadableFiles.file_purpose,
+                        "id",
+                        DownloadableFiles.id,
+                    )
+                ),
+            )
+            .filter(TrialMetadata.trial_id == DownloadableFiles.trial_id)
+            .group_by(TrialMetadata)
```

Comment:
if you either way doing group_by here,
why not make it to final view (the one that `_build_file_bundle` makes now) right here with SQL group by `type` and `purpose` ?